kernel/common/enumerable.rb

```diff
@@ -212,6 +212,12 @@ def slice_when(&block)
     end
   end
 
+  def chunk_while(&block)
+    raise ArgumentError, "wrong number of arguments (0 for 1)" unless block_given?
```

Comment:
Hm, interesting that this doesn't return an Enumerator when no block is given. I see this in the 2.3.0 `ri chunk_while` docs:

```
= .chunk_while

(from ruby core)
=== Implementation from Enumerable
------------------------------------------------------------------------------
  enum.chunk_while {|elt_before, elt_after| bool } -> an_enumerator

------------------------------------------------------------------------------
```

Assuming this should raise an ArgumentError, I agree with @jemc here, we should use a more helpful error message.